File: problems/1823-determine-if-string-halves-are-alike/determine-if-string-halves-are-alike.php
class Solution {

    /**
     * @param String $s
     * @return Boolean
     */
    function halvesAreAlike($s) {
        $vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'];
        $mid = strlen($s) / 2;
        
        $first = substr($s, 0, $mid);
        $second = substr($s, $mid);

        $countFirst = 0;
        $countSecond = 0;
        for ($i = 0; $i < $mid; $i++) {
            if (in_array($first[$i], $vowels)) {
                $countFirst++;
            }
            if (in_array($second[$i], $vowels)) {
                $countSecond++;
            }
        }
        return $countFirst === $countSecond;
    }
}

File: problems/1823-determine-if-string-halves-are-alike/determine-if-string-halves-are-alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']
        mid = int(len(s)/2)
        first = s[0: mid]
        second = s[mid:]
        
        countFirst = 0
        for el in first:
            if el in vowels:
                countFirst += 1
        
        countSecont = 0
        for el in second:
            if el in vowels:
                countSecont += 1
        
        return countFirst == countSecont
        